### ISSUE TITLE
Lint/UnusedMethodArgument AllowUnusedKeywordArguments should default to `true` to be safe

### ISSUE THREAD

**Issue by jdufresne** [rubocop/rubocop#10870]
## Is your feature request related to a problem? Please describe.

By default, `Lint/UnusedMethodArgument` will raise an offense for unused keyword arguments. However, these keyword arguments may be a part of a strict interface in a class hierarchy. 

Removing or renaming the keyword argument may result in an error at the call site.

Relaxing the interface to allow all keyword arguments through sacrifices the intended strictness. Typos or incorrect arguments will go unnoticed.

## Describe the solution you'd like

Make `AllowUnusedKeywordArguments` default `true` as the current default is not safe, IMO.

## Describe alternatives you've considered

I have made this adjustment in my local configuration, so this doesn't prevent me from using the cop. However, I'm advocating for better defaults.

## Additional context

Test case:

```rb
# frozen_string_literal: true

class A
  def f(arg, kwarg: true)
    puts arg
  end
end

class B < A
  def f(arg, kwarg: true)
    super
    puts kwarg
  end
end
```

Result with defaults:

```
Inspecting 1 file
W

Offenses:

test.rb:4:14: W: Lint/UnusedMethodArgument: Unused method argument - kwarg.
  def f(arg, kwarg: true)
             ^^^^^

1 file inspected, 1 offense detected
```

```
$ bundle exec rubocop -V
1.33.0 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 3.1.2 x86_64-linux)
```

**Comment by dvandersluis** [comment]
I don't know if the cop is wrong necessarily though, and your classes could be rewritten as so:

```ruby
class A
  def f(arg)
    puts arg
  end
end

class B < A
  def f(arg, kwarg: true)
    super(arg)
    puts kwarg
  end
end
```

A superclass method isn't forced to have the same argument list as its subclass.

**Comment by jdufresne** [comment]
> A superclass method isn't forced to have the same argument list as its subclass.

I agree, it isn't required by Ruby.

I also agree that I could rewrite particular lines make the violation go away. 

However, strict interfaces are often benefit large codebases as calling `*.f` is predictable at all points in the class hierarchy. One does not need to know whether the instance is a parent class or a child class to safely call it. It removes complexity at the call site.

I think the option should stick around, just have its default flipped for the typical case.

**Comment by dvandersluis** [comment]
I agree with you that there can be benefits to having the same argument list everywhere. However, equally valid arguments can be made for both styles, depending on the style you prefer.

I recommend opening an issue with [rubocop/ruby-style-guide](https://github.com/rubocop/ruby-style-guide) and if it is agreed upon there that the default should be to allow unused kwargs we can certainly change the default on rubocop.
